include/swift/AST/DiagnosticsSema.def

```diff
@@ -4690,6 +4690,10 @@ ERROR(objc_invalid_on_static_subscript,none,
 ERROR(objc_invalid_with_generic_params,none,
       "%0 cannot be %" OBJC_ATTR_SELECT "1 because it has generic parameters",
       (DescriptiveDeclKind, unsigned))
+ERROR(objc_invalid_with_generic_requirements,none,
+      "%0 cannot be %" OBJC_ATTR_SELECT "1 because it has generic requirements "
```

Comment:
How about "because it has a 'where' clause"? "Generic requirements" is vague, and saying "that cannot be represented in Objective-C" suggests that _some_ requirements can be represented, which is not the case.